main.go

```diff
@@ -128,3 +133,60 @@ func registerCapabilities(port string, log logger.Handler) {
 		log.Info(err.Error())
 	}
 }
+
+func registerDynamicCapabilities(port string, log logger.Handler) {
+	registerWorkloads(port, log)
+	//Start the ticker
+	const reRegisterAfter = 24
+	ticker := time.NewTicker(reRegisterAfter * time.Hour)
+	for {
+		<-ticker.C
+		registerWorkloads(port, log)
+	}
+
+}
+
+func registerWorkloads(port string, log logger.Handler) {
+	appVersion, chartVersion, err := getLatestValidAppVersionAndChartVersion()
+	if err != nil {
+		fmt.Println("Could not get latest version")
+		return
+	}
+	log.Info("Registering latest workload components")
+	// Register workloads
+	if err := adapter.RegisterWorkLoadsDynamically(mesheryServerAddress(), serviceAddress()+":"+port, &adapter.DynamicComponentsConfig{
+		TimeoutInMinutes: 10,
+		URL:              "https://github.com/kumahq/charts/releases/download/kuma-" + chartVersion + "/kuma-" + chartVersion + ".tgz",
+		GenerationMethod: adapter.HelmCHARTS,
+		Config: manifests.Config{
+			Name:        smp.ServiceMesh_Type_name[int32(smp.ServiceMesh_KUMA)],
+			MeshVersion: appVersion,
+			Filter: manifests.CrdFilter{
+				RootFilter:    []string{"$[?(@.kind==\"CustomResourceDefinition\")]"},
+				NameFilter:    []string{"$..[\"spec\"][\"names\"][\"kind\"]"},
+				VersionFilter: []string{"$..spec.versions[0]", " --o-filter", "$[0]"},
+				GroupFilter:   []string{"$..spec", " --o-filter", "$[]"},
+				SpecFilter:    []string{"$..openAPIV3Schema.properties.spec", " --o-filter", "$[]"},
+			},
+		},
+		Operation: config.KumaOperation,
+	}); err != nil {
+		log.Info(err.Error())
+		return
+	}
+	log.Info("Latest workload components successfully registered.")
+}
+func getLatestValidAppVersionAndChartVersion() (string, string, error) {
```

Comment:
Not sure about the internals of the helm chart function but if we are only concerned about the latest version of the chart then `>0.0.0-0` does just that. No need to query the latest chart.